docs/shared/cypress-plugin.md

```diff
@@ -83,6 +83,101 @@ nx e2e frontend-e2e --baseUrl=https://frontend.com
 
+## Code coverage
+1. instrument cypress of your `frontend-e2e` and your frontend framework to use code coverage as described on the official cypress documentation. See below how to insctruct angular. You may also need to check if the setting `pluginsFile` of `frontend-e2e/cypress.json` needs to be updated.
+
+2. Add a `nyc` example configuration (you can adjust the configuration later on according to your needs) to your e2e-app (in this example in the file `apps/frontend-e2e/.nycrc`):
+```json
+{
+  "temp-dir": "../../.nyc_output",
+  "report-dir": "../../.coverage",
+  "reporter": ["html"],
+  "clean": true
+}
+```
+Most probably you will want to ignore `.nyc_output` and `.coverage` adding them to `.gitignore`.
+
+3. Add a `nyc` example configuration (you can adjust this later on according tou your needs) to the file `.nycrc` to make a report in the root directory:
+```json
+{ 
+  "reporter": ["html"],
+  "skip-full": true,
+  "per-file": true
+}
+```
+
+4. Install `nyc` as a development dependency if not yet done.
+5. Now run the e2e test for `frontend-e2e` as usual.
+6. Execute the reporter for quick info in the terminal like this:
+```bash
+npx nyc report -r text-summary
+```
+7. In `.coverage` you will also find the html report.
+
+Amongst others you can also check the coverage and throw error when it is not hight enough using `nyc`. See the official documentation of nyc.
+
+### For Angular developers
+instrument angular to use code coverage as following:
+
+1. Install `@jsdevtools/coverage-istanbul-loader`
+```bash
+yarn add --dev @jsdevtools/coverage-istanbul-loader
+```
+```bash
+npm install --save-dev @jsdevtools/coverage-istanbul-loader
+```
+
+2. Create a custom webpack file `coverage.webpack.js` with the following content:
+
+```javascript
+const path = require('path');
+
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.(js|ts|html)$/,
+        loader: '@jsdevtools/coverage-istanbul-loader',
+        options: { esModules: true },
+        enforce: 'post',
+        exclude: [
+          /\.(e2e|spec)\.ts$/,
+          /node_modules/,
+          /(ngfactory|ngstyle)\.js/,
+        ],
+      },
+    ],
+  },
+};
+```
+
+3. Add [custom webpack support](guides/customize-webpack) in the `angular.json` or as probably the case may be in `apps/example-app/project.json` using `coverage.webpack.js` above. To complete this create a new target `serve-coverage` by copying the `serve` target. Then add the webpack file above to the options and replace the executor (currently the executor you need is `ngx-build-plus:dev-server`). In the snippet below you see the changed lines. The rest stays as is:
```

Comment:
You can just say in `in your projects configuration` instead of talking about the potential configuration files. and then link to the page where project configs are talked about here https://nx.dev/configuration/projectjson#configuration-projectjson-and-nxjson